av1an-core/src/encoder.rs

```diff
@@ -232,44 +232,118 @@ impl Encoder {
   }
 
   /// Returns default settings for the encoder
-  pub fn get_default_arguments(self) -> Vec<String> {
+  pub fn get_default_arguments(self, tiles: (u32, u32)) -> Vec<String> {
     match self {
       // aomenc automatically infers the correct bit depth, and thus for aomenc, not specifying
       // the bit depth is actually more accurate because if for example you specify
       // `--pix-format yuv420p`, aomenc will encode 10-bit when that is not actually the desired
       // pixel format.
-      Encoder::aom => into_vec![
-        "--threads=8",
-        "--cpu-used=6",
-        "--end-usage=q",
-        "--cq-level=30",
-        "--tile-columns=2",
-        "--tile-rows=1",
-      ],
-      Encoder::rav1e => into_vec![
-        "--tiles",
-        "8",
-        "--speed",
-        "6",
-        "--quantizer",
-        "100",
-        "--no-scene-detection",
-      ],
+      Encoder::aom => {
+        let defaults: Vec<String> = into_vec![
+          "--threads=8",
+          "--cpu-used=6",
+          "--end-usage=q",
+          "--cq-level=30",
+        ];
+
+        match tiles.0 > 1 || tiles.1 > 1 {
```

Comment:
Doing `match` on a bool may be confusing to other people looking at this code. I think doing an `if ... else` would be clearer and would avoid needing to write the match arms. (same for the other instances below)

Reply:
ok